api/src/permissions/modules/process-ast/lib/field-map-from-ast.ts

```diff
@@ -0,0 +1,14 @@
+import type { SchemaOverview } from '@directus/types';
+import type { AST } from '../../../../types/ast.js';
+import type { FieldMap } from '../types.js';
+import { extractFieldsFromChildren } from './extract-fields-from-children.js';
+import { extractFieldsFromQuery } from './extract-fields-from-query.js';
+
+export function fieldMapFromAst(ast: AST, schema: SchemaOverview): FieldMap {
+	const fieldMap: FieldMap = new Map();
+
+	extractFieldsFromChildren(ast.name, ast.children, fieldMap, schema);
+	extractFieldsFromQuery(ast.name, ast.query, fieldMap, schema);
```

Comment:
Alias fields should be accounted for here as invalid fields are currently returned, causing errors when the path existence is validated. 🤔

Reply:
Fixed in https://github.com/directus/directus/pull/22384/commits/a1870749aa34808f19285d3423cf6016cc343b41